### PR TITLE
Provided correct enum members, method overloads

### PR DESCRIPTION
**Pull request by rpetrusha**
# Provided correct enum members, method overloads

## Summary

This may have been a migration problem. I've fixed two major issues:

- A number of items in [Obsolete Members in the .NET Framework](https://docs.microsoft.com/en-us/dotnet/framework/whats-new/obsolete-members) referenced only the type and not the member. This was particularly true of enumeration members. 
- A number of items failed to specify the particular method overload that was deprecated.
 
## Suggested Reviewers

@mairaw

### CODE REVIEW

**Comment by mairaw** [comment]
I'll start my review. But I did double check a few in the last version before the migration and the enumeration values were there. @wh-alice @yishengjin1413 this is another topic where the enum values were not migrated properly. Is there something else we can do to cover the missed enum values? Thanks!

File: docs/framework/whats-new/obsolete-members.md [hunk 1]
```diff
@@ -224,14 +224,14 @@ The tables in this article list the type members that are obsolete in the .NET F
 |<xref:System.Security.Permissions.FileIOPermissionAttribute?displayProperty=nameWithType>|<xref:System.Security.Permissions.FileIOPermissionAttribute.All%2A>|Please use the <xref:System.Security.Permissions.FileIOPermissionAttribute.ViewAndModify%2A?displayProperty=nameWithType> property instead.|
```
Comment (mairaw) [inline docs/framework/whats-new/obsolete-members.md]: line 204 should link to System.Security.HostSecurityManagerOptions.HostPolicyLevel

File: docs/framework/whats-new/obsolete-members.md [hunk 2]
```diff
@@ -224,14 +224,14 @@ The tables in this article list the type members that are obsolete in the .NET F
...
-|<xref:System.Security.Permissions.ReflectionPermissionFlag?displayProperty=nameWithType>|<xref:System.Security.Permissions.ReflectionPermissionFlag>|This permission is no longer used by the CLR.|  
-|<xref:System.Security.Permissions.ReflectionPermissionFlag?displayProperty=nameWithType>|<xref:System.Security.Permissions.ReflectionPermissionFlag>|This API has been deprecated.|  
+|<xref:System.Security.Permissions.ReflectionPermissionFlag?displayProperty=nameWithType>|<xref:System.Security.Permissions.ReflectionPermissionFlag.AllFlags>|This permission has been deprecated. The <xref:System.Security.Permissions.ReflectionPermissionFlag> enumeration member does not include <xref:System.Security.Permissions.ReflectionPermissionFlag>. Use <xref:System.Security.Permissions.PermissionState?displayProperty=nameWithType> to get full access.|
```
Comment (mairaw) [inline docs/framework/whats-new/obsolete-members.md]: third xref should be AllFlags too and 4th one should be System.Security.Permissions.ReflectionPermissionFlag.RestrictedMemberAccess and 5th should be System.Security.Permissions.PermissionState.Unrestricted

File: docs/framework/whats-new/obsolete-members.md [hunk 3]
```diff
@@ -224,14 +224,14 @@ The tables in this article list the type members that are obsolete in the .NET F
...
-|<xref:System.Security.Permissions.SecurityAction?displayProperty=nameWithType>|<xref:System.Security.Permissions.SecurityAction>|Assembly level declarative security is obsolete and is no longer enforced by the CLR by default.|  
-|<xref:System.Security.Permissions.SecurityAction?displayProperty=nameWithType>|<xref:System.Security.Permissions.SecurityAction>|Assembly level declarative security is obsolete and is no longer enforced by the CLR by default.|  
-|<xref:System.Security.Permissions.SecurityAction?displayProperty=nameWithType>|<xref:System.Security.Permissions.SecurityAction>|Assembly level declarative security is obsolete and is no longer enforced by the CLR by default.|  
+|<xref:System.Security.Permissions.SecurityAction?displayProperty=nameWithType>|<xref:System.Security.Permissions.SecurityAction.Deny>|<xref:System.Security.Permissions.SecurityAction> is obsolete and will be removed in a future release of the .NET Framework.|
```
Comment (mairaw) [inline docs/framework/whats-new/obsolete-members.md]: 3rd xref link should be Deny too

File: docs/framework/whats-new/obsolete-members.md [hunk 4]
```diff
@@ -329,7 +329,7 @@ The tables in this article list the type members that are obsolete in the .NET F
   
 |Type|Member|Message|  
 |----------|------------|-------------|  
-|<xref:System.Diagnostics.Eventing.Reader.StandardEventKeywords?displayProperty=nameWithType>|<xref:System.Diagnostics.Eventing.Reader.StandardEventKeywords>|First deprecated in the .NET Framework 4.5.<br /><br /> Incorrect value; use <xref:System.Diagnostics.Eventing.Reader.StandardEventKeywords> instead.|  
+|<xref:System.Diagnostics.Eventing.Reader.StandardEventKeywords?displayProperty=nameWithType>|<xref:System.Diagnostics.Eventing.Reader.StandardEventKeywords.CorrelationHint>|First deprecated in the .NET Framework 4.5.<br /><br /> Incorrect value; use <xref:System.Diagnostics.Eventing.Reader.StandardEventKeywords> instead.|
```
Comment (mairaw) [inline docs/framework/whats-new/obsolete-members.md]: last xref should be System.Diagnostics.Eventing.Reader.StandardEventKeywords.CorrelationHint2

File: docs/framework/whats-new/obsolete-members.md [hunk 5]
```diff
@@ -496,13 +496,13 @@ The tables in this article list the type members that are obsolete in the .NET F
 |<xref:System.ComponentModel.MemberDescriptor?displayProperty=nameWithType>|<xref:System.ComponentModel.MemberDescriptor.GetInvokee%2A>|This method has been deprecated. Use <xref:System.ComponentModel.MemberDescriptor.GetInvocationTarget%2A?displayProperty=nameWithType> instead.|
```
Comment (mairaw) [inline docs/framework/whats-new/obsolete-members.md]: lines 377, 378, 380, 381, 383, 384, 387, 388 last link should be System.Security.Permissions.PermissionState.None

File: docs/framework/whats-new/obsolete-members.md [hunk 6]
```diff
@@ -496,13 +496,13 @@ The tables in this article list the type members that are obsolete in the .NET F
...
-|<xref:System.ComponentModel.Design.SelectionTypes?displayProperty=nameWithType>|<xref:System.ComponentModel.Design.SelectionTypes>|This value has been deprecated. Use <xref:System.Enum?displayProperty=nameWithType> class methods to determine valid values, or use a type converter.|  
-|<xref:System.ComponentModel.Design.ViewTechnology?displayProperty=nameWithType>|<xref:System.ComponentModel.Design.ViewTechnology>|This value has been deprecated. Use <xref:System.ComponentModel.Design.ViewTechnology?displayProperty=nameWithType> instead.|  
-|<xref:System.ComponentModel.Design.ViewTechnology?displayProperty=nameWithType>|<xref:System.ComponentModel.Design.ViewTechnology>|This value has been deprecated. Use <xref:System.ComponentModel.Design.ViewTechnology?displayProperty=nameWithType> instead.|  
+|<xref:System.ComponentModel.Design.SelectionTypes?displayProperty=nameWithType>|<xref:System.ComponentModel.Design.SelectionTypes.Click>|This value has been deprecated. Use <xref:System.ComponentModel.Design.SelectionTypes?displayProperty=nameWithType> instead.|
```
Comment (mairaw) [inline docs/framework/whats-new/obsolete-members.md]: last link should be System.ComponentModel.Design.SelectionTypes.Primary

File: docs/framework/whats-new/obsolete-members.md [hunk 7]
```diff
@@ -496,13 +496,13 @@ The tables in this article list the type members that are obsolete in the .NET F
...
+|<xref:System.ComponentModel.Design.SelectionTypes?displayProperty=nameWithType>|<xref:System.ComponentModel.Design.SelectionTypes.Click>|This value has been deprecated. Use <xref:System.ComponentModel.Design.SelectionTypes?displayProperty=nameWithType> instead.|  
+|<xref:System.ComponentModel.Design.SelectionTypes?displayProperty=nameWithType>|<xref:System.ComponentModel.Design.SelectionTypes.MouseDown>|This value has been deprecated. It is no longer supported.|  
+|<xref:System.ComponentModel.Design.SelectionTypes?displayProperty=nameWithType>|<xref:System.ComponentModel.Design.SelectionTypes.MouseUp>|This value has been deprecated. It is no longer supported.|  
+|<xref:System.ComponentModel.Design.SelectionTypes?displayProperty=nameWithType>|<xref:System.ComponentModel.Design.SelectionTypes.Normal>|This value has been deprecated. Use <xref:System.ComponentModel.Design.SelectionTypes?displayProperty=nameWithType> instead.|
```
Comment (mairaw) [inline docs/framework/whats-new/obsolete-members.md]: last link should be System.ComponentModel.Design.SelectionTypes.Auto

File: docs/framework/whats-new/obsolete-members.md [hunk 8]
```diff
@@ -496,13 +496,13 @@ The tables in this article list the type members that are obsolete in the .NET F
...
+|<xref:System.ComponentModel.Design.SelectionTypes?displayProperty=nameWithType>|<xref:System.ComponentModel.Design.SelectionTypes.MouseUp>|This value has been deprecated. It is no longer supported.|  
+|<xref:System.ComponentModel.Design.SelectionTypes?displayProperty=nameWithType>|<xref:System.ComponentModel.Design.SelectionTypes.Normal>|This value has been deprecated. Use <xref:System.ComponentModel.Design.SelectionTypes?displayProperty=nameWithType> instead.|  
+|<xref:System.ComponentModel.Design.SelectionTypes?displayProperty=nameWithType>|<xref:System.ComponentModel.Design.SelectionTypes.Valid>|This value has been deprecated. Use <xref:System.Enum?displayProperty=nameWithType> class methods to determine valid values, or use a type converter.|  
+|<xref:System.ComponentModel.Design.ViewTechnology?displayProperty=nameWithType>|<xref:System.ComponentModel.Design.ViewTechnology.Passthrough>|This value has been deprecated. Use <xref:System.ComponentModel.Design.ViewTechnology?displayProperty=nameWithType> instead.|
```
Comment (mairaw) [inline docs/framework/whats-new/obsolete-members.md]: last link should be System.ComponentModel.Design.ViewTechnology.Default

File: docs/framework/whats-new/obsolete-members.md [hunk 9]
```diff
@@ -496,13 +496,13 @@ The tables in this article list the type members that are obsolete in the .NET F
...
+|<xref:System.ComponentModel.Design.SelectionTypes?displayProperty=nameWithType>|<xref:System.ComponentModel.Design.SelectionTypes.Normal>|This value has been deprecated. Use <xref:System.ComponentModel.Design.SelectionTypes?displayProperty=nameWithType> instead.|  
+|<xref:System.ComponentModel.Design.SelectionTypes?displayProperty=nameWithType>|<xref:System.ComponentModel.Design.SelectionTypes.Valid>|This value has been deprecated. Use <xref:System.Enum?displayProperty=nameWithType> class methods to determine valid values, or use a type converter.|  
+|<xref:System.ComponentModel.Design.ViewTechnology?displayProperty=nameWithType>|<xref:System.ComponentModel.Design.ViewTechnology.Passthrough>|This value has been deprecated. Use <xref:System.ComponentModel.Design.ViewTechnology?displayProperty=nameWithType> instead.|  
+|<xref:System.ComponentModel.Design.ViewTechnology?displayProperty=nameWithType>|<xref:System.ComponentModel.Design.ViewTechnology.WindowsForms>|This value has been deprecated. Use <xref:System.ComponentModel.Design.ViewTechnology?displayProperty=nameWithType> instead.|
```
Comment (mairaw) [inline docs/framework/whats-new/obsolete-members.md]: last link should be System.ComponentModel.Design.ViewTechnology.Default

File: docs/framework/whats-new/obsolete-members.md [hunk 10]
```diff
@@ -516,9 +516,9 @@ The tables in this article list the type members that are obsolete in the .NET F
...
-|<xref:System.Diagnostics.EventLogPermissionAccess?displayProperty=nameWithType>|<xref:System.Diagnostics.EventLogPermissionAccess>|This member has been deprecated. Please use <xref:System.Diagnostics.EventLogPermissionAccess?displayProperty=nameWithType> instead.|  
-|<xref:System.Diagnostics.EventLogPermissionAccess?displayProperty=nameWithType>|<xref:System.Diagnostics.EventLogPermissionAccess>|This member has been deprecated. Please use <xref:System.Diagnostics.EventLogPermissionAccess?displayProperty=nameWithType> instead.|  
-|<xref:System.Diagnostics.EventLogPermissionAccess?displayProperty=nameWithType>|<xref:System.Diagnostics.EventLogPermissionAccess>|This member has been deprecated. Please use <xref:System.Diagnostics.EventLogPermissionAccess?displayProperty=nameWithType> instead.|  
+|<xref:System.Diagnostics.EventLogPermissionAccess?displayProperty=nameWithType>|<xref:System.Diagnostics.EventLogPermissionAccess.Audit>|This member has been deprecated. Please use <xref:System.Diagnostics.EventLogPermissionAccess?displayProperty=nameWithType> instead.|
```
Comment (mairaw) [inline docs/framework/whats-new/obsolete-members.md]: last link should be System.Diagnostics.EventLogPermissionAccess.Administer

File: docs/framework/whats-new/obsolete-members.md [hunk 11]
```diff
@@ -516,9 +516,9 @@ The tables in this article list the type members that are obsolete in the .NET F
...
-|<xref:System.Diagnostics.EventLogPermissionAccess?displayProperty=nameWithType>|<xref:System.Diagnostics.EventLogPermissionAccess>|This member has been deprecated. Please use <xref:System.Diagnostics.EventLogPermissionAccess?displayProperty=nameWithType> instead.|  
-|<xref:System.Diagnostics.EventLogPermissionAccess?displayProperty=nameWithType>|<xref:System.Diagnostics.EventLogPermissionAccess>|This member has been deprecated. Please use <xref:System.Diagnostics.EventLogPermissionAccess?displayProperty=nameWithType> instead.|  
+|<xref:System.Diagnostics.EventLogPermissionAccess?displayProperty=nameWithType>|<xref:System.Diagnostics.EventLogPermissionAccess.Audit>|This member has been deprecated. Please use <xref:System.Diagnostics.EventLogPermissionAccess?displayProperty=nameWithType> instead.|  
+|<xref:System.Diagnostics.EventLogPermissionAccess?displayProperty=nameWithType>|<xref:System.Diagnostics.EventLogPermissionAccess.Browse>|This member has been deprecated. Please use <xref:System.Diagnostics.EventLogPermissionAccess?displayProperty=nameWithType> instead.|
```
Comment (mairaw) [inline docs/framework/whats-new/obsolete-members.md]: last link should be System.Diagnostics.EventLogPermissionAccess.Administer

File: docs/framework/whats-new/obsolete-members.md [hunk 12]
```diff
@@ -516,9 +516,9 @@ The tables in this article list the type members that are obsolete in the .NET F
...
-|<xref:System.Diagnostics.EventLogPermissionAccess?displayProperty=nameWithType>|<xref:System.Diagnostics.EventLogPermissionAccess>|This member has been deprecated. Please use <xref:System.Diagnostics.EventLogPermissionAccess?displayProperty=nameWithType> instead.|  
+|<xref:System.Diagnostics.EventLogPermissionAccess?displayProperty=nameWithType>|<xref:System.Diagnostics.EventLogPermissionAccess.Audit>|This member has been deprecated. Please use <xref:System.Diagnostics.EventLogPermissionAccess?displayProperty=nameWithType> instead.|  
+|<xref:System.Diagnostics.EventLogPermissionAccess?displayProperty=nameWithType>|<xref:System.Diagnostics.EventLogPermissionAccess.Browse>|This member has been deprecated. Please use <xref:System.Diagnostics.EventLogPermissionAccess?displayProperty=nameWithType> instead.|  
+|<xref:System.Diagnostics.EventLogPermissionAccess?displayProperty=nameWithType>|<xref:System.Diagnostics.EventLogPermissionAccess.Instrument>|This member has been deprecated. Please use <xref:System.Diagnostics.EventLogPermissionAccess?displayProperty=nameWithType> instead.|
```
Comment (mairaw) [inline docs/framework/whats-new/obsolete-members.md]: last link should be System.Diagnostics.EventLogPermissionAccess.Write

File: docs/framework/whats-new/obsolete-members.md [hunk 13]
```diff
@@ -551,7 +551,7 @@ The tables in this article list the type members that are obsolete in the .NET F
 |<xref:System.Net.HttpWebRequest?displayProperty=nameWithType>|<xref:System.Net.HttpWebRequest.%23ctor>|First deprecated in the .NET Framework 4.5.<br /><br /> Use of this member generates a compiler error.<br /><br /> This API supports the .NET Framework infrastructure and is not intended to be used directly from your code.|
```
Comment (mairaw) [inline docs/framework/whats-new/obsolete-members.md]: line 530 - second xref link should be System.Diagnostics.PerformanceCounterPermissionAccess.Browse and second xref link should be System.Diagnostics.PerformanceCounterPermissionAccess.Read

line 531 - second xref link should be System.Diagnostics.PerformanceCounterPermissionAccess.Instrument and second xref link should be System.Diagnostics.PerformanceCounterPermissionAccess.Write

File: docs/framework/whats-new/obsolete-members.md [hunk 14]
```diff
@@ -710,8 +710,8 @@ The tables in this article list the type members that are obsolete in the .NET F
```
Comment (mairaw) [inline docs/framework/whats-new/obsolete-members.md]: line 688 - second xref link should be System.Windows.Forms.AccessibleStates.Valid

File: docs/framework/whats-new/obsolete-members.md [hunk 15]
```diff
@@ -710,8 +710,8 @@ The tables in this article list the type members that are obsolete in the .NET F
   
 |Type|Member|Message|  
 |----------|------------|-------------|  
-|<xref:System.Xml.ValidationType?displayProperty=nameWithType>|<xref:System.Xml.ValidationType>|Validation type should be specified as <xref:System.Xml.ValidationType?displayProperty=nameWithType> or <xref:System.Xml.ValidationType?displayProperty=nameWithType>.|  
-|<xref:System.Xml.ValidationType?displayProperty=nameWithType>|<xref:System.Xml.ValidationType>|XDR validation through <xref:System.Xml.XmlValidatingReader?displayProperty=nameWithType> is obsoleted.|  
+|<xref:System.Xml.ValidationType?displayProperty=nameWithType>|<xref:System.Xml.ValidationType.Auto>|Validation type should be specified as <xref:System.Xml.ValidationType?displayProperty=nameWithType> or <xref:System.Xml.ValidationType?displayProperty=nameWithType>.|
```
Comment (mairaw) [inline docs/framework/whats-new/obsolete-members.md]: 3rd xref link should be System.Xml.ValidationType.DTD and 4th link should be System.Xml.ValidationType.Schema

**Review by mairaw — CHANGES_REQUESTED** [review]
@rpetrusha you did an excellent job on restoring some of the missing values. I carefully went through our last DDUEML version and added a few other missing ones. I'm also wondering if we should review the links which should have been autoUpgrade = false.

File: docs/framework/whats-new/obsolete-members.md [hunk 16]
```diff
@@ -131,9 +131,9 @@ The tables in this article list the type members that are obsolete in the .NET F
...
 |<xref:System.Enum?displayProperty=nameWithType>|<xref:System.Enum.ToString(System.IFormatProvider)>|The `provider` argument is not used. Please use <xref:System.Enum.ToString%2A?displayProperty=nameWithType>.|  
-|<xref:System.Enum?displayProperty=nameWithType>|<xref:System.Enum.ToString%2A>|The `provider` argument is not used. Please use <xref:System.Enum.ToString%2A?displayProperty=nameWithType>.|  
-|<xref:System.LoaderOptimization?displayProperty=nameWithType>|<xref:System.LoaderOptimization>|This method has been deprecated. Please use <xref:System.Reflection.Assembly.Load%2A?displayProperty=nameWithType> instead.|  
-|<xref:System.LoaderOptimization?displayProperty=nameWithType>|<xref:System.LoaderOptimization>|This method has been deprecated. Please use <xref:System.Reflection.Assembly.Load%2A?displayProperty=nameWithType> instead.|  
+|<xref:System.Enum?displayProperty=nameWithType>|<xref:System.Enum.ToString(System.String,System.IFormatProvider)>|The `provider` argument is not used. Please use <xref:System.Enum.ToString%2A?displayProperty=nameWithType>.|
```
[the remark this answers is not in the document]
Comment (rpetrusha) [reply docs/framework/whats-new/obsolete-members.md]: @mairaw, no it doesn't. It should link to a specific overload. I'll change these.

File: docs/framework/whats-new/obsolete-members.md (hunk 1 above)
Comment (rpetrusha) [reply docs/framework/whats-new/obsolete-members.md]: Thanks for catching this, @mairaw.

**Comment by rpetrusha** [comment]
I've incorporated all of your comments, @mairaw, as well as corrected some additional issues. Thanks for doing such a thorough review!

File: docs/framework/whats-new/obsolete-members.md (hunk 6 above)
Comment (mairaw) [reply docs/framework/whats-new/obsolete-members.md]: @rpetrusha this one wasn't changed

File: docs/framework/whats-new/obsolete-members.md (hunk 7 above)
Comment (mairaw) [reply docs/framework/whats-new/obsolete-members.md]: same here @rpetrusha

**Comment by rpetrusha** [comment]
Sorry that I missed those, @mairaw. I think I've addressed all of your comments now.

**Review by mairaw — APPROVED** [review]

**Pull request merged**